src/SDKs/BotService/Microsoft.Azure.Management.BotService/Customizations/AzureBotServiceClient.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Microsoft.Azure.Management.BotService.Customizations;
+
+namespace Microsoft.Azure.Management.BotService
+{
+    /// <summary>
+    /// Azure Bot Service is a platform for creating smart conversational
+    /// agents.
+    /// </summary>
+    public partial class AzureBotServiceClient
+    {
+        /// <summary>
+        /// Azure Active Directory TenantId
+        /// </summary>
+        public string TenantId { get; set; }
+
+        partial void CustomInitialize()
+        {
+            if(BotServiceConfiguration.ShouldProvisionMsaApp)
```

Comment:
@carlosscastro will the user even need to have non-prod bot env.? Or is this for test purpose?